File: src/main/java/nl/chimpgamer/networkmanager/extensions/autoban/models/PunishmentAction.java
package nl.chimpgamer.networkmanager.extensions.autoban.models;

import lombok.AllArgsConstructor;
import lombok.Getter;
import nl.chimpgamer.networkmanager.api.models.punishments.Punishment;

@AllArgsConstructor
@Getter
public class PunishmentAction {
    private Punishment.Type onActionType;
    private int count;
    private Punishment.Type actionType;
    private long duration;
    private String reason;
}

File: src/main/java/nl/chimpgamer/networkmanager/extensions/autoban/models/PunishmentAction.kt
package nl.chimpgamer.networkmanager.extensions.autoban.models

import nl.chimpgamer.networkmanager.api.models.punishments.Punishment

data class PunishmentAction(val onActionType: Punishment.Type, val count: Int = 0, val actionType: Punishment.Type, val duration: Long = 0, val reason: String)